packages/oats-runtime/src/make.ts

```diff
@@ -438,6 +438,12 @@ export function makeObject<
     if (typeof value !== 'object' || value == null || Array.isArray(value)) {
       return getErrorWithValueMsg('expected an object', value);
     }
+
+    // if the value been parsed already using the given object type
+    // return the value as is
+    if (type && getTypeSet(value)?.has(type)) {
+      return Make.ok(value);
+    }
     comparisorOrder ||= Object.keys(props);
```

Comment:
how will this work with `oneOf`s? specially in the case where a discriminator cannot be determined, wouldn't we still check each class until we find the one that the object was already an instance of?

Reply:
we would check each branch of the oneOf -- however because of line 444 the checks should be fast. 

Or do I miss your point completely?

Reply:
ah. now I got it. Yes we would still try to parse in each branch which is not great. We could add the oneOf type to the metadata for the parsed object to sidestep that I think 🤔 . Kinda would be *way* better to have discriminated objects as that would help in more cases.